NDTensors/src/UnallocatedArrays/src/unallocated_impl.jl

```diff
@@ -0,0 +1,29 @@
+for Typ in (:UnallocatedFill, :UnallocatedZeros)
+  ## Here are functions specifically defined for UnallocatedArrays
+  ## not implemented by FillArrays
+  @eval begin
+    alloctype(A::$Typ) = alloctype(typeof(A))
+    alloctype(::Type{<:$Typ{ElT,N,Axes,Alloc}}) where {ElT,N,Axes,Alloc} = Alloc
+
+    get_index(A::$Typ) = getindex(data(A))
+
+    array(A::$Typ) = alloctype(typeof(A))(data(A))
+    Array(A::$Typ) = array(A)
+    ## A function for NDTensors to launch functions off of
+    is_immutable(A::$Typ) = is_immutable(typeof(A))
+    is_immutable(::Type{<:$Typ}) = true
+  end
+  ## TODO I don't think this is the correct way to call
+  ## functions which are defined in `FillArrays`
+  for fun in (:size, :length, :convert, :sum, :getindex_value)
+    @eval FillArrays.$fun(A::$Typ) = $fun(data(A))
+  end
+  ## TODO Here I am defining LinearAlgebra functions in one sweep
+  for fun in (:norm,)
+    @eval LinearAlgebra.$fun(A::$Typ) = $fun(data(A))
+  end
+end
+
+function copy(F::UnallocatedFill)
```

Comment:
I think you can just return `F` since it is immutable, so copying doesn't need to do anything (copying just protects against mutation, but you can't mutate it anyway).

Reply:
My thought was that you might want a newly constructed object with copy. say we do
```
A = Tensor(UnallocatedFill(2, (10,20)))
B = copy(A)

A = ## Change the values of A requiring it to allocate
GC.gc()
```
What happens to the fill from `A`? Will it be thrown away or is the compiler smart enough to keep it around?

Reply:
Well, you can't modify `UnallocatedFill` in-place so `A = ...` will always make a new object.

Reply:
I realize I was answering a slightly different question.

In the situation you laid out, the garbage collector would  know that `B` is still referring to the original tensor that was first bound to `A`, even if the copy is a no-op.

Reply:
Got it (As I am sure you know, in C++ this can be an issue because you might deconstruct the fill and `B` might still be pointing to is which can cause memory issues). Thanks!